src/file/ShareDialog.tsx

```diff
@@ -60,25 +60,21 @@ const labelStyles = mergeStyleSets({
 
 const ShareText: React.FC = () => {
     const { scene } = useScene();
-    const [data, setData] = useState<string | undefined>(undefined);
-    const [copyMessageVisible, setMessageVisibility] = useState(false);
-    const [timeout, setMessageTimeout] = useState<number>();
-
-    useEffect(() => {
-        (async () => setData(await sceneToText(scene)))();
-    }, [scene]);
+    const data = useMemo(() => sceneToText(scene), [scene]);
+    const [copyMessageVisible, setMessageVisibility] = useBoolean(false);
+    const timerRef = useRef<number>();
 
     const doCopyToClipboard = () => {
-        navigator.clipboard.writeText(data || '');
-        setMessageVisibility(true);
-        clearTimeout(timeout);
-        setMessageTimeout(setTimeout(() => setMessageVisibility(false), 2000));
+        navigator.clipboard.writeText(data);
+        setMessageVisibility.setTrue();
+        clearTimeout(timerRef.current);
+        timerRef.current = setTimeout(() => setMessageVisibility.setFalse(), 2000);
     };
     const labelClasses = `${labelStyles.message} ${copyMessageVisible ? '' : labelStyles.hidden}`;
 
-    if (!data) {
-        return <Spinner />;
-    }
+    useEffect(() => {
+        return () => clearTimeout(timerRef.current);
+    }, [scene]);
```

Comment:
Does this effect need to be re-evaluated when the scene changes? It doesn't use the scene variable inside it anywhere.

Reply:
Ah, that's correct - that was a holdover from when `useEffect` was used to generate the Plan Code. I've removed it from the PR.